src/java/org/apache/cassandra/utils/AbstractGuavaIterator.java

```diff
@@ -14,23 +16,25 @@
  * limitations under the License.
  */
 
-package org.apache.cassandra.index.sasi.utils;
+package org.apache.cassandra.utils;
 
 import java.util.NoSuchElementException;
 
 import com.google.common.collect.PeekingIterator;
 
 import static com.google.common.base.Preconditions.checkState;
 
-// This is fork of the Guava AbstractIterator, the only difference
-// is that state & next variables are now protected, this was required
-// for SkippableIterator.skipTo(..) to void all previous state.
-public abstract class AbstractIterator<T> implements PeekingIterator<T>
+/**
+ * This is fork of the Guava AbstractIterator, the only difference
+ * is that state & next variables are now protected, this was required
+ * for SkippableIterator.skipTo(..) to void all previous state.
```

Comment:
nit: I guess `SkippableIterator` doesn't exist anymore, but even the basic `RangeIterator` uses `next` in `skipTo()`.  Also, `state` can actually be `private` again.